app/components/forms/wizard/sponsors-step.js

```diff
@@ -58,16 +61,53 @@ export default Component.extend(FormMixin, {
         this.sendAction('save');
       });
     },
-    move(direction) {
+    savePublished() {
       this.onValid(() => {
-        this.sendAction('move', direction);
+        this.set('data.event.state', 'published');
+        this.sendAction('save', this.data);
       });
     },
-    publish() {
+    move(direction) {
       this.onValid(() => {
-        this.set('data.event.state', 'published');
-        this.sendAction('save');
+        this.sendAction('move', direction);
       });
+    },
+    openConfirmModal() {
+      this.set('isPublishUnpublishModalOpen', true);
+    },
+
+    togglePublishState() {
```

Comment:
This is still repeated

Reply:
sponsor and attendee can't inherit EventWizard

Reply:
only basis-step-detail, other-details, session-speaker can inherit. On inheriting in sponsors-step.js, And attendee it is causing error and other functionalities are not working as expected.

Reply:
You can create another mixin and inherit that in those steps. There is always a way to remove duplication